src/engine.ts

```diff
@@ -360,7 +398,7 @@ export function initializePolyfill() {
   function getOrCreateInstance(node: Node): Instance {
     let instance = getInstance(node);
     if (!instance) {
-      let innerController: NodeController<Node>;
+      let innerController: NodeController<Node|ShadowRoot>;
```

Comment:
`ShadowRoot` should inherit from `Node` so this shouldn't be necessary anymore

Reply:
good call out, I wasn't sure about that bit when I added it.